openstack_plugin_common/__init__.py

```diff
@@ -212,9 +218,16 @@ def use_external_resource(ctx, sugared_client, openstack_type,
                           name_field_name='name'):
     if not is_external_resource(ctx):
         return None
-
-    resource = _get_resource_by_name_or_id_from_ctx(
-        ctx, name_field_name, openstack_type, sugared_client)
+    resource = None
+    try:
+        resource = _get_resource_by_name_or_id_from_ctx(
+            ctx, name_field_name, openstack_type, sugared_client)
+    except NonRecoverableError:
+        if not resource and is_create_if_missing(ctx):
```

Comment:
unnecessary 'if not resource and...' 
can check only-
'if is_create_if_missing(ctx):'
...
else
...